### PR TITLE
Jocelyn dev 13

### PR DESCRIPTION
**Pull request by jocelynshen**

### CODE REVIEW

**Review by clarisaleu — APPROVED** [review]

File: app/src/main/java/com/example/footprnt/Map/MapFragment.java [hunk 1]
```diff
@@ -463,7 +463,7 @@ public void onClick(View v) {
                 mSwipe.start();                 // Add sound when user sends post
                 final String title = ((EditText) mAlertDialog.findViewById(R.id.etTitle)).getText().toString();
                 final String snippet = ((EditText) mAlertDialog.findViewById(R.id.etSnippet)).getText().toString();
-                if( TextUtils.isEmpty(title) || TextUtils.isEmpty(snippet)){
+                if( TextUtils.isEmpty(title) || TextUtils.isEmpty(snippet)) {
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/MapFragment.java:466]: Why a space at the start of the paranthesis?

if(TextUtils.isEmpty(title) || TextUtils.isEmpty(snippet)) {

File: app/src/main/java/com/example/footprnt/Map/StreetViewActivity.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import android.os.Bundle;
+import android.support.v4.app.FragmentActivity;
+
+import com.example.footprnt.R;
+import com.google.android.gms.maps.OnStreetViewPanoramaReadyCallback;
+import com.google.android.gms.maps.StreetViewPanorama;
+import com.google.android.gms.maps.StreetViewPanoramaFragment;
+import com.google.android.gms.maps.model.LatLng;
+
+public class StreetViewActivity extends FragmentActivity implements OnStreetViewPanoramaReadyCallback {
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/StreetViewActivity.java:12]: Javadoc please

File: app/src/main/java/com/example/footprnt/Map/StreetViewActivity.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    private LatLng mLatLng;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Double lat = getIntent().getExtras().getDouble("latitude");
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/StreetViewActivity.java:19]: To be safe I'd do some null checks here to see if getExtras() is null

File: app/src/main/java/com/example/footprnt/Map/Util/MapConstants.java [hunk 4]
```diff
@@ -31,9 +31,10 @@
     public static BitmapDescriptor DEFAULT_MARKER = BitmapDescriptorFactory.defaultMarker(BitmapDescriptorFactory.HUE_BLUE);
 
     // Filter menu items
-    public static String MENU_ITEMS[] = {"Create", "Current", "Discover", "View"};
+    public static String MENU_ITEMS[] = {"Create", "Current", "Discover", "View", "Street"};
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/Util/MapConstants.java:34]: I'd still like to see these as resources so they can be translated if needed.

File: app/src/main/java/com/example/footprnt/Map/Util/MapUtil.java [hunk 5]
```diff
@@ -207,18 +60,18 @@ public static JSONObject getContinents(Activity activity){
...
-    public static void geoLocate(EditText searchText, GoogleMap map, Context context){
+    public static void geoLocate(EditText searchText, GoogleMap map, Context context) {
         String searchString = searchText.getText().toString();
 
         Geocoder geocoder = new Geocoder(context);
         List<Address> list = new ArrayList<>();
         try{
             list = geocoder.getFromLocationName(searchString, 1);
-        } catch (IOException e){
+        } catch (IOException e) {
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/Util/MapUtil.java:70]: We should probably do something in this catch block

File: app/src/main/java/com/example/footprnt/Map/Util/MapUtil.java [hunk 6]
```diff
@@ -207,18 +60,18 @@ public static JSONObject getContinents(Activity activity){
...
 
         }
 
-        if (list.size() > 0){
+        if (list.size() > 0) {
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/Util/MapUtil.java:74]: if(!list.isEmpty()) {

File: app/src/main/java/com/example/footprnt/Map/Util/ServerUtil.java [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class ServerUtil {
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/Util/ServerUtil.java:30]: Javadoc please :)

File: app/src/main/java/com/example/footprnt/Map/Util/ServerUtil.java [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+        newPost.setTitle(title);
+        newPost.setLocation(new ParseGeoPoint(point.latitude, point.longitude));
+        Geocoder gcd = new Geocoder(context, Locale.getDefault());
+        List<Address> addresses = null;
+        try {
+            addresses = gcd.getFromLocation(point.latitude, point.longitude, 1);
+            if (addresses.size() > 0) {
+                String city = addresses.get(0).getLocality();
+                String country = addresses.get(0).getCountryName();
+                String country_code = addresses.get(0).getCountryCode();
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/Util/ServerUtil.java:63]: nit:
countryCode

File: app/src/main/java/com/example/footprnt/Map/Util/UiUtil.java [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+public class UiUtil {
+
+    /**
+     * Calculates relative time
+     *
+     * @param rawJsonDate input date to convert
+     * @return relative time
+     */
+    public static String getRelativeTimeAgo(String rawJsonDate) {
+        String twitterFormat = "EEE MMM dd HH:mm:ss ZZZZZ yyyy";
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/Util/UiUtil.java:40]: This never changes so make it a class constant

File: app/src/main/java/com/example/footprnt/Map/Util/UiUtil.java [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+     * @param post post to retrieve tags from
+     * @return string format of tag (ie. #culture #food)
+     */
+    public static String getPostTags(Post post) {
+        String tagname = "";
+        JSONArray arr = post.getTags();
+        if (arr != null && arr.length() > 0) {
+            for (int i = 0; i < arr.length(); i++) {
+                try {
+                    tagname += "#" + arr.getString(i) + " ";
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/Util/UiUtil.java:83]: StringBuilder would be much more efficient here.

File: app/src/main/java/com/example/footprnt/Map/Util/UiUtil.java [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+
+    /**
+     * Handles displaying all images into the view holder
+     *
+     * @param post    post to display
+     * @param holder  holder to display images into
+     * @param context current context of post
+     */
+    public static void setPostImages(Post post, PostAdapter.ViewHolder holder, Context context) {
+        if(post.getImage()!=null) {
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/Util/UiUtil.java:143]: if(post.getImage() != null) {

File: app/src/main/java/com/example/footprnt/Map/Util/UiUtil.java [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+            Glide.with(context).load(imgUrl).into(holder.ivPicture);
+            holder.ivPicture.setVisibility(View.VISIBLE);
+        } else {
+            holder.ivPicture.setVisibility(View.GONE);
+        }
+        if (post.getUser().getParseFile("profileImg") != null) {
+            String userImgUrl = post.getUser().getParseFile("profileImg").getUrl();
+            Glide.with(context).load(userImgUrl).apply(RequestOptions.circleCropTransform()).into(holder.ivUserPicture);
+        } else {
+            Glide.with(context).load("http://via.placeholder.com/300.png").apply(RequestOptions.circleCropTransform()).into(holder.ivUserPicture);
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/Util/UiUtil.java:154]: "http://via.placeholder.com/300.png" should be a constant

**Pull request merged**